### ISSUE TITLE
Why does my alpaca-lora model repeat the content of the question when generating?

### ISSUE THREAD

**Issue by li-aolong** [tloen/alpaca-lora#524]
Besides the content of the question, there are always some statements about the results. For example, 

**Instruction: Write a Python program that prints the first 10 Fibonacci numbers.**

**tloen/alpaca-lora-7b**
```
Flock, lock, rock, sock, and nock.
```
**my_alpaca-lora-7b**
```
Here are five words that rhyme with 'shock': 

1. Rock
2. Lock
3. Dock
4. Cock
5. Hock.
```

**Instruction: List all Canadian provinces in alphabetical order.**

**tloen/alpaca-lora-7b**
```
Alberta, British Columbia, Manitoba, New Brunswick, Newfoundland and Labrador, Northwest Territories, Nova Scotia, Nunavut, Ontario, Prince Edward Island, Quebec, Saskatchewan
```
**my_alpaca-lora-7b**
```
Here is a list of all Canadian provinces in alphabetical order:

1. Alberta
2. British Columbia
3. Manitoba
4. New Brunswick
5. Newfoundland and Labrador
6. Northwest Territories
7. Nova Scotia
8. Nunavut
9. Ontario
10. Prince Edward Island
11. Quebec
12. Saskatchewan
13. Yukon.
```

**Instruction: Write a Python program that prints the first 10 Fibonacci numbers.**

**tloen/alpaca-lora-7b**
```
def fibonacci(n):
    if n == 0:
        return 0
    elif n == 1:
        return 1
    else:
        return fibonacci(n-1) + fibonacci(n-2)

for i in range(1,11):
```
**my_alpaca-lora-7b**
``````
Here is a Python program that prints the first 10 Fibonacci numbers:

```python
def fibonacci(n):
    if n == 1:
        return 1
    elif n == 2:
        return 1
    else:
        return fibonacci(n - 1) + fibonacci(n - 2)

for i in range(1, 11):
    print(fibonacci(i))
```

This program defines a function `fibonacci` that takes an integer `n` as an argument and returns the `n`th Fibonacci number. If `n` is less than or equal to 1, the function returns 1. If `n` is equal to 2, the function returns 1. Otherwise, the function calls itself recursively with `n - 1` and `n - 2` and returns the sum of the two results.

The `for` loop iterates through the first 10 Fibonacci numbers from 1 to 10. For each number, the function `fibonacci` is called with the
``````

my training script on 2 A100 40G:
```
WORLD_SIZE=2 CUDA_VISIBLE_DEVICES=0,1 torchrun --nproc_per_node=2 --master_port=2345 \
    finetune.py \
    --base_model '~/models/llama-7b' \
    --data_path '~/codes/alpaca-lora/datas' \
    --batch_size 128 \
    --micro_batch_size 32 \
    --num_epochs 10 \
    --learning_rate 3e-4 \
    --cutoff_len 512 \
    --val_set_size 2000 \
    --lora_r 16 \
    --lora_alpha 16 \
    --lora_dropout 0.05 \
    --lora_target_modules '[q_proj,k_proj,v_proj,o_proj]' \
    --train_on_inputs \
    --group_by_length
```

my loss curve:
![image](https://github.com/tloen/alpaca-lora/assets/12003487/a18aac09-412c-4acf-b626-131c49fdb40b)
![image](https://github.com/tloen/alpaca-lora/assets/12003487/c79d8247-abb6-40f6-8c0d-5dea8ce0b840)
I use the best checkpoint at 1600 step

**Comment by mriganktiwari** [comment]
I too have fine-tuned a Llama-7b with exact same parameters and data as given [here](https://github.com/tloen/alpaca-lora#training-finetunepy), and got a adapter_weights saved.

- When I use these adapter weights saved through  my training, I get something similar as you say, **it keeps repeating the ##Input section of the prompt, without doing anything said in the instruction**.
- This output is very different from what I get when using the adapter weights from HF [repo](https://huggingface.co/tloen/alpaca-lora-7b)

Am unable to understand how can this differ?

**Comment by evelynxyx** [comment]
> I too have fine-tuned a Llama-7b with exact same parameters and data as given [here](https://github.com/tloen/alpaca-lora#training-finetunepy), and got a adapter_weights saved.
> 
> * When I use these adapter weights saved through  my training, I get something similar as you say, **it keeps repeating the ##Input section of the prompt, without doing anything said in the instruction**.
> * This output is very different from what I get when using the adapter weights from HF [repo](https://huggingface.co/tloen/alpaca-lora-7b)
> 
> Am unable to understand how can this differ?

I face the same problem
